examples/example_otap.py

```diff
@@ -35,72 +56,131 @@
+                                                "5_days"],
+                        help="OTAP action delay")
+
+    # Log parameters
+    parser.add_argument("--log-level", default="info", type=str,
+                        choices=["debug", "info", "warning", "error", "critical"],
+                        help="Default to 'info'. Log level to be displayed. "
+                        "It has to be chosen between 'debug', 'info', 'warning', 'error' and 'critical'")
+    # Script behavior
+    parser.add_argument("--strict-mode",
+                        dest='strict_mode',
+                        action='store_true',
+                        help="Stop execution at first generated error on gateway/sink operation")
     args = parser.parse_args()
 
-    logging.basicConfig(format='%(levelname)s %(asctime)s %(message)s', level=logging.WARNING)
+    logging.basicConfig(format='%(asctime)s | [%(levelname)s] %(filename)s:%(lineno)d:%(message)s', level=args.log_level.upper(),
+                        handlers=[
+                            logging.StreamHandler(sys.stdout),
+                            logging.FileHandler("example_otap.log", mode="w")
+                        ]
+    )
+
+    logging.debug(f"Script arguments: {args}")
 
     wni = WirepasNetworkInterface(args.host,
                                   args.port,
                                   args.username,
                                   args.password,
-                                  insecure=args.insecure)
-
-    if args.network is None:
-        print("No network address provided")
-        exit()
+                                  insecure=args.insecure,
+                                  strict_mode=args.strict_mode)
+
+    if args.cmd == "delayed" or args.cmd == "update_delay":
+        if args.otap_delay is None:
+            logging.error("No value given for OTAP action with delay")
+            exit()
+        else:
+            delay = otap_delay_choices_to_internal_values[args.otap_delay]
+            delay_str = args.otap_delay.replace('_',' ')
+
+    # Get gateway list
+    gateways = None
+    if args.gateway is None:
+        if args.gateway_from_file is not None:
+            # Load gateway list from file
+            gateways = []
+            try:
+                for line in args.gateway_from_file:
+                    gateways.append(line.strip('\n'))
+            except ValueError:
+                logging.error("Invalid file format. Must be UTF-8 with lines ending with LF character")
+                exit()
+    else:
+        gateways = args.gateway
+
+    if gateways is None:
+       logging.info("OTAP to be performed on all gateways under network %d" % args.network)
+    else:
+        logging.info("OTAP to be performed on %d gateways" % len(gateways))
+        logging.debug(f"Gateway list {gateways}")
 
     otapHelper = WirepasOtapHelper(wni,
                                    args.network,
                                    args.gateway)
```

Comment:
It should be `gateways` instead of `args.gateway` I think

Reply:
Indeed...